components/bookshelf/detailedbook/utils.ts

```diff
@@ -0,0 +1,22 @@
+import { BookType } from "api/types"
+
+export const checkRating = (book: BookType) => {
+  if (book.rating) {
+    return book.rating
+  }
+  return 0
+}
+
+export const checkID = (book: BookType) => {
+  if (book.user_book_id) {
+    return book.user_book_id
+  }
+  return null
+}
+
+export const checkFinishDate = (book: BookType) => {
+  if (book.finish_date) {
```

Comment:
better return a value instead of `true` since the `if` function will count it as truthy

```
return book.finish_date ? book.finish_date : false # can just be book.finish_date, but I think you're probably encountering a problem if that's used in itself
```

Reply:
good idea. implemented this